### PR TITLE
F5 framework selector

### PR DESCRIPTION
**Pull request by BillHiebert**
Sorry, I needed to recreate the branch which closed the original PR https://github.com/dotnet/roslyn-project-system/pull/1585. 

This should just contain the changes to address the code review feedback.

PR for post dev15 RTW.
Implemented the design that was available in Dev14:
Added a new menu item to the debug dropdown which shows the currently active framework being used for debugging.
a. The selected framework appears in the Framework menu name so that you don’t have to expand the menu to see the active one.
b. The frameworks appear in the menu in the order they are defined in the property
c. The current active framework is remembered in the .user file so it persists across VS sessions. It defaults to the first one if there is no existing setting or the existing setting doesn’t match any the frameworks being targeted.
d. The framework names are displayed as they exist in the project file. I’d like to display something better but I didn’t have any code to convert the short name to a friendly name. We could possible resolve to the TargetFrameworkMoniker property and display that - but that requires more evaluation. Certainly something to address later.
Added an IActiveDebugFrameworkServices public interface which exposes access to the ConfiguredProject for the active debug framework.
Updated the Debugger providers and (token replacer) to resolve properties against the active debug frameworks ConfiguredProject.
Unit test coverage for the new components
Fixed a problem with ProjectExportProvider. It claims it can be called on a background thread but doing so deadlocked as it was using QueryService to get to ProjectServices. Fixed it to just import the ProjectServices MEF component.

### CODE REVIEW

File: src/ProjectSystem.sln [hunk 1]
```diff
@@ -3,61 +3,61 @@ Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
 VisualStudioVersion = 15.0.26226.0
 MinimumVisualStudioVersion = 10.0.40219.1
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "DeployTestDependencies", "DeployTestDependencies\DeployTestDependencies.csproj", "{37BA82E6-9ABD-4ACA-AA26-2DFD39A359A5}"
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{A50EC328-5D30-4624-98AD-B649461DB772}"
```
Comment (basoundr) [inline src/ProjectSystem.sln]: @BillHiebert Are these changes required?

Comment (BillHiebert) [inline src/ProjectSystem.sln]: No. Not sure why this change is here. I need to revert it

**Comment by BillHiebert** [comment]
Opened https://github.com/dotnet/roslyn-project-system/issues/1674 to track fixing the framework names in the menu

**Comment by kzu** [comment]
Is this available/going to be available in the preview channel for VS2017? If so, when?
